dffml/util/net.py

```diff
@@ -195,12 +211,26 @@ def cached_download_unpack_archive(
     >>> asyncio.run(files_in_dffml_commit_c4469a())
     124
     """
+
+    def on_error(func, path, exc_info):
+        """
+        If there's permission denied error. Error handler for shutil.rmtree
+        """
+        if not os.access(path, os.W_OK):
+            os.chmod(path, stat.S_IWUSR)
+            func(path)
+        else:
+            raise DirectoryNotExtractedError(path)
+
```

Comment:
Per comment below

```suggestion
```

Reply:
Why we are removing this on_error method?